packages/plugin-core/src/utils/editor.ts

```diff
@@ -236,3 +246,48 @@ export async function isBrokenWikilink(): Promise<boolean> {
   );
   return type === DECORATION_TYPES.brokenWikilink;
 }
+
+export type ProcessSelectionOpts = {
+  qs?: string;
+  vault?: DVault;
+  anchor?: DNoteAnchorBasic;
+  overrides?: Partial<NoteProps>;
+  kind?: TargetKind;
+  /**
+   * What {@link vscode.ViewColumn} to open note in
+   */
+  column?: ViewColumn;
+  /** added for contextual UI analytics. */
+  source?: string;
+};
+
+/**
+ * NOTE: this method requires that `ExtensionProvider` be available and can provide a workspace
+ */
+export async function getLinkFromSelectionWithWorkspace() {
+  const { selection, editor } = VSCodeUtils.getSelection();
+  if (
+    _.isEmpty(selection) ||
+    _.isUndefined(selection) ||
+    _.isUndefined(selection.start) ||
```

Comment:
nitpick: you could simplify the above 3 conditions as `selection?.start !== undefined`

Reply:
this is needed so that typescript compiler doesn't complain about selection being undefined in following blocks. added comments